pkg/services/ngalert/state/manager.go

```diff
@@ -463,7 +463,7 @@ func (st *Manager) setNextState(ctx context.Context, alertRule *ngModels.AlertRu
 	//
 	// This is temporary change to make sure that the labels are not persistent in the state after it was in Error state
 	// TODO yuri. Remove it when correct Error result with labels is provided
-	if currentState.State == eval.Error && result.State != eval.Error {
+	if (currentState.State == eval.Error || currentState.State == eval.Alerting) && result.State != eval.Error {
```

Comment:
Maybe with `currentState.State == eval.Alerting` we should also check that `rule.ExecErrState == models.AlertingErrState`? Not sure